server/app-engine/static/shapepoint.js

```diff
@@ -1,7 +1,48 @@
 class ShapePoint {
-    constructor(lat, lon) {
-        this.lat = lat;
-        this.lon = lon;
+    constructor(lat = null, lon = null) {
+        util.log("ShapePoint constructor");
+        if(lat !== null){
+            this.lat = parseFloat(lat);
+        } else this.lat = lat;
+        if(lon !== null){
+            this.lon = parseFloat(lon);
+        } else this.lon = lon;
+    }
+
+    setLatIfLess(lat){
+        let latFloat = parseFloat(lat);
```

Comment:
The args passed into these functions appear to be strings, whereas in the python version they are numerical. Port should work the same as original wherever possible